src/js/utils/linked-list.js

```diff
@@ -107,10 +113,11 @@ export default class LinkedList {
   forEach(callback) {
     let item = this.head;
     let index = 0;
+    let nextItem;
     while (item) {
-      callback(item, index);
-      index++;
-      item = item.next;
+      nextItem = item.next;
+      callback(item, index++);
```

Comment:
Would prefer a `changeTolerantForEach` or something. Or a flag. This change makes `forEach` behave less like it would with an array `for` loop, and thus is a pretty surprising quirk.

We had delete-safe iteration in a few places of the codebase before, we must have.